apps/admin-gui-e2e/src/integration/trusted-facility-admin-perun_admin.spec.js

```diff
@@ -0,0 +1,259 @@
+      .get('[data-cy=managers]')
+      .click()
+      .get('[data-cy=add-manager-button]')
+      .click()
+      .get('[data-cy=search-manager-input]')
+      .type(`${dbUserToAssignRoles}`)
+      .get('[data-cy=search-manager-button]')
+      .click()
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .click()
+      .intercept('**/authzResolver/getRichAdmins**').as('getRichAdmins')
+      .get('[data-cy=add-manager-button-dialog]')
+      .click()
+      .wait('@getRichAdmins')
+
+      // assert that manager was added
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .should('exist')
+  })
+
+  it('test add resource observer', () => {
+
+    cy.get('[data-cy=advanced-settings]')
+      .click()
+      .get('[data-cy=managers]')
+      .click()
+      .get('[data-cy=role-filter]')
+      .click()
+      .get('[data-cy=resourceobserver]')
+      .click()
+      .get('[data-cy=add-manager-button]')
+      .click()
+      .get('[data-cy=search-manager-input]')
+      .type(`${dbUserToAssignRoles}`)
+      .get('[data-cy=search-manager-button]')
+      .click()
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .click()
+      .intercept('**/authzResolver/getRichAdmins**').as('getRichAdmins')
+      .get('[data-cy=add-manager-button-dialog]')
+      .click()
+      .wait('@getRichAdmins')
+
+      // assert that manager was added
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .should('exist')
+  })
+
+  it('test add resource self service', () => {
+
+    cy.get('[data-cy=advanced-settings]')
+      .click()
+      .get('[data-cy=managers]')
+      .click()
+      .get('[data-cy=role-filter]')
+      .click()
+      .get('[data-cy=resourceselfservice]')
+      .click()
+      .get('[data-cy=add-manager-button]')
+      .click()
+      .get('[data-cy=search-manager-input]')
+      .type(`${dbUserToAssignRoles}`)
+      .get('[data-cy=search-manager-button]')
+      .click()
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .click()
+      .intercept('**/authzResolver/getRichAdmins**').as('getRichAdmins')
+      .get('[data-cy=add-manager-button-dialog]')
+      .click()
+      .wait('@getRichAdmins')
+
+      // assert that manager was added
+      .get(`[data-cy=${dbUserToAssignRoles}-checkbox]`)
+      .should('exist')
+  })
+
+  it('test remove resource manager', () => {
+
+    cy.get('[data-cy=advanced-settings]')
+      .click()
+      .get('[data-cy=managers]')
+      .click()
+      .get(`[data-cy=${dbUserToRemoveRoles}-checkbox]`)
+      .click()
+      .get('[data-cy=remove-manager-button]')
+      .click()
+      .intercept('**/authzResolver/getRichAdmins**').as('getRichAdmins')
+      .get('[data-cy=remove-manager-button-dialog]')
+      .click()
+      .wait('@getRichAdmins')
+
+      // assert that manager was removed
+      .get(`[data-cy=${dbUserToRemoveRoles}-checkbox]`)
+      .should('not.exist')
+  })
+
+  it('test remove resource observer', () => {
```

Comment:
You should change the role filter at the end of this test because now you are checking that `Test8 User17` is not Resource Admin NOT Resource Observer. After that, you should move intercept() and wait() functions after the role filter change.